lib/resources/etc_hosts.rb

```diff
@@ -0,0 +1,82 @@
+# encoding: utf-8
+# author: Matthew Dromazos
+
+require 'utils/parser'
+require 'pry'
+class EtcHosts < Inspec.resource(1)
+  name 'etc_hosts'
+  desc 'Use the etc_hosts InSpec audit resource to find an
+    ip_address and its associated hosts'
+  example "
+    describe etc_hosts.where { ip_address == '127.0.0.1' } do
+      its ( 'ip_address' ) { should cmp '127.0.0.1' }
+      its ( 'primary_name' ) { should cmp 'localhost' }
+      its ( 'all_host_names' ) { should eq [['localhost', 'localhost.localdomain', 'localhost4', 'localhost4.localdomain4']] }
+    end
+  "
+
+  attr_reader :params
+
+  include CommentParser
+
+  def initialize(hosts_path = nil)
+    return skip_resource 'The `etc_hosts` resource is not supported on your OS.' unless inspec.os.linux? || inspec.os.windows?
+    @conf_path      = hosts_path || default_hosts_file_path(hosts_path)
```

Comment:
You're doing an `||` here (which is great), which means `default_hosts_file_path` doesn't need to receive it as an argument. It will simplify the `default_hosts_file_path` method as well.